chain/jsonrpc-adversarial-primitives/Cargo.toml

```
@@ -0,0 +1,23 @@
+[package]
+name = "near-jsonrpc-adversarial-primitives"
+version = "0.2.0"
+authors = ["Near Inc <hello@nearprotocol.com>"]
+publish = false
+edition = "2018"
+
+[dependencies]
+serde = { version = "1", features = ["derive"], optional = true }
+serde_json = { version = "1", optional = true }
+near-primitives = { path = "../../core/primitives" }
+
+near-jsonrpc-primitives = { path = "../jsonrpc-primitives", optional = true }
+near-network = { path = "../network" }
+
+
+[features]
+ser_de = [
+  "serde_json",
+  "serde",
```

I feel this might work:

```suggestion
serde = [
  "serde_json",
```

---

Unfortunately, cargo doesn't support that yet. You have to explicitly include the dependency in the feature list.

The closest alternative would be to rename the crate to free up the feature name.

Example: https://github.com/miraclx/nearcore/commit/e499e195f7dfaf57ca5d1871c9b5ee22f8e6c7a0